src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx

```diff
@@ -29,62 +32,150 @@ import ExternalReference from 'src/shared/components/external-reference/External
 import { getEntityViewerSidebarPayload } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
 
 import { RootState } from 'src/store';
-import { EntityViewerSidebarPayload } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState';
-import { ExternalLink } from 'src/content/app/entity-viewer/types/externalLink';
+import {
+  CrossReference,
+  CrossReferenceGroup
+} from 'src/content/app/entity-viewer/types/crossReference';
+import { EntityViewerParams } from 'src/content/app/entity-viewer/EntityViewer';
 
 import styles from './GeneExternalReferences.scss';
 
-type Props = {
-  sidebarPayload: EntityViewerSidebarPayload | null;
+const QUERY = gql`
+  query Gene($stable_id: String!, $genome_id: String!) {
+    gene(byId: { stable_id: $stable_id, genome_id: $genome_id }) {
+      name
+      stable_id
+      cross_references {
+        id
+        name
+        description
+        url
+        source {
+          id
+          name
+          url
+        }
+      }
+      transcripts {
+        stable_id
+        cross_references {
+          id
+          name
+          description
+          url
+          source {
+            id
+            name
+            url
+          }
+        }
+      }
+    }
+  }
+`;
+
+type Gene = {
+  name: string;
+  stable_id: string;
+  transcripts: {
+    stable_id: string;
+    cross_references: CrossReference[];
+  }[];
+  cross_references: CrossReference[];
 };
 
-const GeneExternalReferences = (props: Props) => {
-  if (!props.sidebarPayload) {
+const getFormattedCrossReferences = (crossReferences: CrossReference[]) => {
+  const geneCrossReferences: { [key: string]: CrossReferenceGroup } = {};
+
+  crossReferences.forEach((xref) => {
+    const sourceId = xref.source.id;
+
+    if (!geneCrossReferences[sourceId]) {
+      geneCrossReferences[sourceId] = {
+        source: xref.source,
+        references: []
+      };
+    }
+
+    geneCrossReferences[sourceId].references.push({
+      id: xref.id,
+      url: xref.url,
+      name: xref.name,
+      description: xref.description
+    });
+  });
+
+  return geneCrossReferences;
+};
+
+const GeneExternalReferences = () => {
+  const params: EntityViewerParams = useParams();
+
+  const entityId = params.entityId?.split(':').pop();
+
+  // TODO: The genomeId is temporarily hardcoded here as Thoas does ot have date for homo_sapiens_GCA_000001405_27.
+  const { data, loading } = useQuery<{ gene: Gene }>(QUERY, {
+    variables: {
+      stable_id: entityId,
+      genome_id: 'homo_sapiens_GCA_000001405_28'
+    },
+    skip: !entityId
+  });
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!data || !data.gene) {
     return <div>No data to display</div>;
   }
+
+  const geneCrossReferences = getFormattedCrossReferences(
```

Comment:
`geneCrossReferences` -> `crossReferenceGroups`?

Reply:
That's done